webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go

```diff
@@ -349,28 +352,48 @@ func (v validator) validateStorageClass(ctx *context.WebhookRequestContext, vm *
 	scPath := field.NewPath("spec", "storageClass")
 	scName := vm.Spec.StorageClass
 
-	// TODO: This validation shouldn't be done in the webhook.
-
-	sc := &storagev1.StorageClass{}
-	if err := v.client.Get(ctx, client.ObjectKey{Name: scName}, sc); err != nil {
-		return append(allErrs, field.Invalid(scPath, scName, fmt.Sprintf(storageClassNotFoundFmt, vm.Namespace)))
+	var zones []topologyv1.AvailabilityZone
+	if pkgconfig.FromContext(ctx).Features.PodVMOnStretchedSupervisor {
+		azs, err := topology.GetAvailabilityZones(ctx, v.client)
```

Comment:
Is it possible to cache and save AZs in the webhook once retrieved, otherwise every time this validation will invoke API to get AZs.

Reply:
Its a simple API call to query the AZ list which is limited, and it will be called on every VM create which isn't a very frequent operation. Plus, in the entire VMOp codebase, we haven't cached the AZ list. Should not be an issue even at scale.